File: hp-server/src/main/java/net/hserver/hplite/message/LocalInnerWear.java
package net.hserver.hplite.message;

import cn.hutool.crypto.digest.MD5;
import cn.hutool.json.JSONUtil;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class LocalInnerWear {

    /**
     * 穿透服务器IP
     */
    private String serverIp;

    /**
     * 穿透服务器的端口
     */
    private Integer serverPort;

    /**
     * 本地IP
     */
    private String localIp;

    /**
     * 本地的端口
     */
    private Integer localPort;

    /**
     * 本地穿透的key
     */
    private String configKey;

    /**
     * 穿透类型
     */
    private ConnectType connectType;

    private String md5;

    public String getMd5() {
        return MD5.create().digestHex(serverIp + serverPort + localIp + localPort + configKey + connectType.name());
    }

    /**
     * 转换为json
     *
     * @return
     */
    public String toJson() {
        return JSONUtil.toJsonStr(this);
    }
}

File: hp-server-golang/bean/local_inner_wear.go
package bean

type LocalInnerWear struct {
	/**
	 * 映射服务器IP
	 */
	ServerIp string `json:"serverIp"`

	/**
	 * 映射服务器的端口
	 */
	ServerPort int `json:"serverPort"`

	/**
	 * 远端端口
	 */
	RemotePort int `json:"remotePort"`

	/**
	 * 本地地址
	 */
	LocalAddress string `json:"localAddress"`

	/**
	 * 本地映射的key
	 */
	ConfigKey string `json:"configKey"`

	/**
	 * 链接方式
	 */
	TunType string `json:"tunType"`

	/**
	 * 内网速度
	 */
	InLimit int `json:"inLimit"`

	/**
	 * 外网速度
	 */
	OutLimit int `json:"outLimit"`
}
